server/samples/clients/node/appencryption_client.js

```diff
@@ -0,0 +1,284 @@
+            self.#encryptResolve = resolve;
+            self.#call.write({encrypt: {data: Buffer.from(payload),},});
+        });
+    }
+
+    /**
+     * Decrypt the data using the current session and its partition
+     * @param drr
+     * @returns {Promise<>}
+     */
+    decrypt(drr) {
+        let self = this;
+        return new Promise(function (resolve, err) {
+            self.#decryptResolve = resolve;
+            self.#call.write({decrypt: {data_row_record: drr,},});
+        });
+    }
+
+    /**
+     * Close the stream
+     */
+    close() {
+        this.#call.end()
+    }
+}
+
+/**
+ * Generate a random string of fixed length
+ * @param length
+ * @returns {string}
+ */
+function randomString(length = 12) {
+    let result = '';
+    let characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+    let charactersLength = characters.length;
+    for (let i = 0; i < length; i++) {
+        result += characters.charAt(Math.floor(Math.random() * charactersLength));
+    }
+
+    return result;
+}
+
+/**
+ * Runs the client
+ * @param client
+ * @returns {Promise<void>}
+ */
+async function run_client(client) {
+    let payload = randomString();
+
+    logger.info(`encrypting payload ${payload}`);
+    let drr = await client.encrypt(payload);
+
+    logger.info(`calling decrypt`);
+    let decryptedPayload = await client.decrypt(drr);
+
+    if (decryptedPayload != payload) {
+        logger.info('oh no... something went terribly wrong!');
+    } else {
+        logger.info(`test completed successfully\n`);
+    }
+}
+
+/**
+ * Executes run_client once
+ * @param client
+ * @returns {Promise<void>}
+ */
+async function run_once(client) {
+    await run_client(client);
+
+    client.close();
+}
+
+/**
+ * Executes run_client until the process is interrupted.
+ * @param client
+ * @returns {Promise<void>}
+ */
+
+async function run_continuously(client) {
+    while (true) {
+        process.on('SIGINT', () => {
+            logger.info('received SIGINT. terminating client');
+            process.exit(0);
+        });
+        process.on('SIGTERM', () => {
+            logger.info('received SIGTERM. terminating client');
+            process.exit(0);
+        });
+        await run_client(client);
+
+        //  Sleep for 1 second before sending the next request
+        sleep.sleep(1);
+    }
+}
+
+/**
+ * Gets a session and calls run_once or run_continuously based on the value of the continuous parameter
+ * @param continuous
+ * @param id
+ * @param callback
+ * @returns {Promise<void>}
+ */
+async function run(continuous, id, appEncryptionProto, callback) {
+    const client = new SessionClient(appEncryptionProto, callback);
+
+    await client.getSession(`partition-${id}`);
+
+    if (continuous) {
+        await run_continuously(client);
+    } else {
+        await run_once(client);
+    }
+}
+
+function loadProtoDef(protoPath) {
+    let appEncryptionDef = protoLoader.loadSync(__dirname + protoPath, {
```

Comment:
This is a bug. `__dirname + protoPath` doesn't work with absolute paths.

For example, given the repo path `/asherah`
Running `node appencryption_client.js --proto-path /asherah/server/protos/appencryption.proto` will result in an error:
```
/asherah/server/samples/clients/node/node_modules/@grpc/proto-loader/node_modules/prot
obufjs/src/root.js:94
            throw err;
            ^

Error: ENOENT: no such file or directory, open '/asherah/server/samples/clients/node/asherah/server/protos/appencryption.proto'
```

Possible fix would be to pass `protoPath` as the filename here and update L261 to use `__dirname`